Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs

```diff
@@ -33,25 +33,64 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Xml;
+using System.Collections.Immutable;
 
 namespace Mono.Addins.Database
 {
 	internal class DatabaseConfiguration
 	{
-		Dictionary<string,AddinStatus> addinStatus = new Dictionary<string,AddinStatus> ();
+		ImmutableDictionary<string, AddinStatus> addinStatus = ImmutableDictionary<string, AddinStatus>.Empty;
 
 		internal class AddinStatus
 		{
-			public AddinStatus (string addinId)
+			public AddinStatus (string addinId, bool configEnabled = false, bool? sessionEnabled = null, bool uninstalled = false, ImmutableArray<string> files = default)
```

Comment:
One thing we could try is using record here, because that gives you implicit immutability with `init-only` properties.

It also provides a base hashcode/iequatable, and mutating via: `original with { SomeProp = newValue }`.

Reply:
Less boilerplate overall :D